### PR TITLE
Update research papers notebook and update the docs

### PR DESCRIPTION
**Pull request by KasukabeDefenceForce**
### :pencil: Description

**Type:** :beetle: `bugfix` | :rocket: `feature` | :biohazard: `breaking change` | :vertical_traffic_light: `testing` | :memo: `documentation` | :roller_coaster: `infrastructure`

This PR aims at improving the logic for fetching the research papers which use Tardis and updating them in the docs.
Old Logic:
We used to search for "Tardis" and "Kerzendorf"  or "tardis" and "supernova" in the full text of the article and were published in the NATUR publication. The total number of articles with this logic were 50.
image
New Logic:
We fetch all the papers which have cited this [article](https://ui.adsabs.harvard.edu/abs/2014MNRAS.440..387K/abstract). The total number of articles fetched with this logic are 132. You can have a look at these [here](https://tardis-sn.github.io/research/papers/)

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by KasukabeDefenceForce** [comment]
I prefer to use requests over the ads package because the former is much faster. I could fetch all the papers in approx 2.5 minutes using requests while it took approx 7 minutes to do the same using ads.

**Comment by KasukabeDefenceForce** [comment]
Sample Workflow Run:
https://github.com/KasukabeDefenceForce/tardis/actions/runs/10387669495

Sample PR:
https://github.com/KasukabeDefenceForce/tardis/pull/44

File: .github/workflows/tardis-research-papers.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    shell: bash -l {0}
+
+env:
+  NASA_ADS_TOKEN: ${{ secrets.NASA_ADS_TOKEN }}
+
+jobs:
+    research-using-tardis:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
```
Comment (atharva-2001) [inline .github/workflows/tardis-research-papers.yml:19]: if you could decrease the indentation it would be nice

**Review by Knights-Templars — APPROVED** [review]

File: .github/workflows/tardis-research-papers.yml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            - name: Setup Python
+              uses: actions/setup-python@v5
+              with:
+                python-version: "3.x"
+    
+            - name: Install dependencies
+              run: |
+                python -m pip install --upgrade pip
+                pip install jupyter nbconvert
+
```
Comment (atharva-2001) [inline .github/workflows/tardis-research-papers.yml:30]: We're using micromamba action, so just for consistency can you try using that?
https://github.com/mamba-org/setup-micromamba#create-environment-from-environment-specs

File: .github/workflows/tardis-research-papers.yml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                path: /tmp
+    
+            - name: Copy files to repository
+              run: |
+                cp -r /tmp/resource_papers/* docs/resources/research_done_using_TARDIS/
+    
+            - name: Get current date
+              run: echo "DATE=$(date +'%Y.%m.%d')" >> $GITHUB_ENV
+    
+            - uses: peter-evans/create-pull-request@v4
```
Comment (atharva-2001) [inline .github/workflows/tardis-research-papers.yml:59]: these actions usually need a pull_request_target trigger- have you tested job this on your fork?

File: .github/workflows/tardis-research-papers.yml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                GITHUB_TOKEN: ${{ secrets.CORE_COORDINATOR_TOKEN }}
+                if: steps.create-pr.outputs.pull-request-operation == 'created'
+    
+            - name: Enable automerge
+              uses: peter-evans/enable-pull-request-automerge@v2
+              with:
+                token: ${{ secrets.INFRASTRUCTURE_COORDINATOR_TOKEN }}
+                pull-request-number: ${{ steps.create-pr.outputs.pull-request-number }}
+                merge-method: squash
+              if: steps.create-pr.outputs.pull-request-operation == 'created'
```
Comment (atharva-2001) [inline .github/workflows/tardis-research-papers.yml:102]: Are we sure we need this?

File: .github/workflows/tardis-research-papers.yml (hunk 3 above)
Comment (KasukabeDefenceForce) [reply .github/workflows/tardis-research-papers.yml:59]: you can check the comments, I have posted my sample runs there

**Review by atharva-2001 — APPROVED** [review]

**Review by jvshields — APPROVED** [review]

**Pull request merged**